src/gui/updater/ocupdater.cpp

```diff
@@ -45,7 +46,27 @@ UpdaterScheduler::UpdaterScheduler(QObject *parent)
         connect(updater, &OCUpdater::newUpdateAvailable,
             this, &UpdaterScheduler::updaterAnnouncement);
 
-        connect(updater, &OCUpdater::requestRestart, this, &UpdaterScheduler::requestRestart);
+        connect(updater, &OCUpdater::requestRestart, this, [this]() {
+            // prevent dialog from being displayed twice (rather unlikely, but it won't hurt)
+            if (_updateFinishedDialog == nullptr) {
+                _updateFinishedDialog = new UpdateFinishedDialog(_parentWidget);
+                _updateFinishedDialog->setAttribute(Qt::WA_DeleteOnClose);
+                _updateFinishedDialog->show();
+                ownCloudGui::raiseDialog(_updateFinishedDialog);
+
+                connect(_updateFinishedDialog, &UpdateFinishedDialog::accepted, this, []() {
+                    if (OC_ENSURE(Utility::isLinux())) {
+                        // restart:
+                        qCInfo(lcUpdater) << "Restarting application NOW, PID" << qApp->applicationPid() << "is ending.";
+                        qApp->quit();
```

Comment:
QTimer::singleShot(0?

Reply:
For the record: this code existed before.